### PR TITLE
KAFKA-17744: improve state updater logs when  restorating state

### PR DESCRIPTION
**Pull request by sebastienviale**
PR linked to https://issues.apache.org/jira/browse/KAFKA-17744

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 1]
```diff
@@ -369,10 +369,15 @@ public static StreamThread create(final TopologyMetadata topologyMetadata,
                                       final Runnable shutdownErrorHook,
                                       final BiConsumer<Throwable, Boolean> streamsUncaughtExceptionHandler) {
 
+        final boolean stateUpdaterEnabled = InternalConfig.stateUpdaterEnabled(config.originals());
+
         final String threadId = clientId + THREAD_ID_SUBSTRING + threadIdx;
+        final String stateUpdaterId = threadId.replace(THREAD_ID_SUBSTRING, STATE_UPDATER_ID_SUBSTRING);
+        final String effectiveThreadId = stateUpdaterEnabled ? stateUpdaterId : threadId;
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: Could you please rename this variable to `restorationThreadId`?

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 2]
```diff
@@ -369,10 +369,15 @@ public static StreamThread create(final TopologyMetadata topologyMetadata,
...
 
         final String logPrefix = String.format("stream-thread [%s] ", threadId);
         final LogContext logContext = new LogContext(logPrefix);
+        final LogContext effectiveLogContext = stateUpdaterEnabled ? new LogContext(String.format("state-updater [%s] ", effectiveThreadId)): logContext ;
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: Similarly, could you please rename this variable to `restorationLogContext`?

**Review by cadonna — COMMENTED** [review]
Thanks for the PR, @sebastienviale !

I had two minor comments.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 1 above)
Comment (sebastienviale) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: done

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 2 above)
Comment (sebastienviale) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: done

**Review by cadonna — APPROVED** [review]
Thanks for the updates, @sebastienviale !

LGTM!

**Pull request merged**